ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/electra/block/BlockProcessorElectra.java

```diff
@@ -522,7 +522,17 @@ protected void addValidatorToRegistry(
     final Validator validator = getValidatorFromDeposit(pubkey, withdrawalCredentials);
     LOG.debug("Adding new validator with index {} to state", state.getValidators().size());
     state.getValidators().append(validator);
-    final int validatorIndex = validatorsUtil.getValidatorIndex(state, pubkey).orElseThrow();
+    int validatorIndex = -1;
+    for (int i = state.getValidators().size() - 1; i >= 0; i--) {
+      if (state.getValidators().get(i).getPublicKey().equals(pubkey)) {
+        validatorIndex = i;
+        break;
+      }
+    }
+    if (validatorIndex < 0) {
```

Comment:
```suggestion
    if (validatorIndex <= 0) {
```
I think this catches a very edge case scenario where pubKey wasn't in the validators set?

Reply:
zero is valid, this is referring back to the initialization of -1... in that case, we didnt find a validator so we'd throw `IllegalStateException`

Reply:
basically a validator index is (theoretically) uint64, and -1 is strictly out of bounds

Reply:
Nevermind my comment, I had another scenario in mind when I was suggesting this.

Reply:
discussed offline, agreement so will resolve this.